text/0814-stylelint.md

```diff
@@ -0,0 +1,138 @@
+
+Prettier itself is already included in the Ember blueprints by default.
+
+### Configuration Changes
+
+A `.stylelintrc.js` file will be added with the following contents:
+
+```js title=".stylelintrc.js"
+'use strict';
+
+module.exports = {
+  extends: ['stylelint-config-standard', 'stylelint-prettier/recommended'],
+};
+```
+
+There are many rules, configs, and plugins available for stylelint and prettier, but we are 
+intentionally avoiding being prescriptive on anything that isn't nearly universally agreed upon 
+by the stylelint, prettier, and Ember communities.
+
+### Blueprint Changes
+
+In general, it is recommended that all blueprints provided by addons should satisfy the default 
+linting configuration of a new Ember application. As such the blueprints provided by ember-cli, 
+ember-source, and ember-data will be updated as needed to ensure that they satisfy these new 
+linting rules requirements.
+
+#### `package.json` scripts
+
+The `app` and `addon` blueprints will be updated to add style-related linting scripts:
+
+```json title="package.json"
+{
+  "scripts": {
+    "lint:style": "stylelint .",
+    "lint:style:fix": "stylelint . --fix",
+  }
+}
+```
+
+## How we teach this
+
+We do not currently discuss linting or stylistic formatting in either guides.emberjs.com or cli.emberjs.com.
+
+However, [previous linting RFCs](https://github.com/emberjs/rfcs/blob/master/text/0628-prettier.md) have 
+proposed adding a subsection to the CLI guides to discuss linting. As that is implemented, we can include 
+basic information about style linting as well.
+
+Documentation for the new rules can be found within the Doc sites & READMEs for the packages being added by this RFC.
+
+## Drawbacks
+
+For those introducing stylelint to an existing codebase, the largest drawback is generally the 
+initial cost of fixing linting violations. This can be mitigated by individually disabling noisy 
+lint rules and working to fix violations overtime, or by extending the [lint-todo](https://github.com/lint-todo/) 
+functionality available for `ember-template-lint` and `eslint` to support `stylelint` as well. 
+However, note that this RFC is more likely to impact newly-generated applications where there 
+will be no existing lint violations to fix as opposed to existing applications.
+
+## Alternatives
+
+>Ember applications may choose to use CSS, SCSS, LESS, or other approaches for styling. How do we 
+>reconcile our default linting config?
+
+Today, the de facto default for Ember apps via the blueprints is CSS, so our default configuration 
+should strive to support that use-case. `stylelint` has readily-available support for popular 
+preprocessors built-in, and there are plugins like [`stylelint-scss`](https://github.com/stylelint-scss/stylelint-scss) 
+that add rules specific to those ecosystems for teams to opt into.
+
+>Why name the scripts `lint:style` instead of `lint:css`?
```

Comment:
I would vote for `lint:css` instead of `lint:style`. Why?

1. The [ember-cli blueprint](https://github.com/ember-cli/ember-cli/blob/master/blueprints/app/files/package.json) already has `lint:js` and `lint:hbs` scripts that are named after the most-common file extension for each. Mixing `lint:style` in with these creates an inconsistency that I'm sure will break people's muscle memory as they try to remember the names of these scripts.
2. The existing `lint:js` script can already lint both JS/TS files, as ESLint is capable of linting both when configured properly. `lint:css` would function the same way in that it could lint different types of CSS files like SCSS/SASS when configured properly.

Reply:
Yea I went back and forth on this and I'm open to community consensus here. I wasn't aware that `lint:js` is considered the standard for TS linting over adding a `lint:ts` command. Do you have thoughts here @chriskrycho?

Reply:
Yeah, we use `lint:js` for all JS-ish files. We use something like `test:types` in general for the type checking pass.

I think `lint:css` makes good sense.

(We’re also going to need to rethink these eventually when first-class component templates become the default, but that’s a little ways out yet and shouldn’t block this at all.)